judge/admin/vote.py

```diff
@@ -0,0 +1,18 @@
+
+from django.contrib import admin
+
+
+class VoteAdmin(admin.ModelAdmin):
+    list_display = ('points', 'voter', 'problem', 'note')
+    search_fields = ('voter', 'problem')
+
+    # if the user has edit all problem or edit own problem perms, so curators authors and superusers
+    def has_change_permission(self, request, obj=None):
```

Comment:
use https://github.com/DMOJ/online-judge/blob/master/judge/models/problem.py#L176 instead

Reply:
so i should def the `is_editable()` function instead?

Reply:
No, I'm saying that your function should call that

Reply:
I don't understand, should my `has_change_permission()` function have the same arguments as `is_editable()` or?

Reply:
What @kiritofeng meant is you just call `obj.problem.is_editable_by` with `request.user`. See [this](https://github.com/DMOJ/online-judge/blob/master/judge/admin/problem.py#L220) for an example.

Reply:
Ah thank you for clarifying.

Reply:
> What @kiritofeng meant is you just call `obj.problem.is_editable_by` with `request.user`. See [this](https://github.com/DMOJ/online-judge/blob/master/judge/admin/problem.py#L220) for an example.

Wait don't I need to add an `is_editable()` in the problempointsvote model then?

Reply:
ah I got it, thank you.